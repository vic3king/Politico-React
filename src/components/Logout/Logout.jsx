import { Component } from 'react';
import authServices from '../../services/authentication.services';

class Logout extends Component {
  componentDidMount() {
    authServices.logout();
  }

  render() {
    return null;
  }
}

export default Logout;
