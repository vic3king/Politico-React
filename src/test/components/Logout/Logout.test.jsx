import React from 'react';
import { shallow } from 'enzyme';
import Logout from '../../../components/Logout/Logout';

describe('Logout component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Logout />);
    expect(wrapper).toMatchSnapshot();
  });

  // it('should render a form tag', () => {
  //   const wrapper = shallow(<Logout />);
  //   expect(wrapper.find('form'));
  //   expect(wrapper.hasClass('signbox2'));
  // });
});
