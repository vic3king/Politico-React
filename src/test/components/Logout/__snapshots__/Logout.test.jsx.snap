// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
